NuRadioMC/simulation/simulation.py

```diff
@@ -754,6 +695,7 @@ def run(self):
                             if(iS is None):
                                 a = 1 / 0
                             electric_field.set_frequency_spectrum(np.array([eR, eTheta, ePhi]), 1. / self._dt)
+                            electric_field = self._raytracer.apply_propagation_effects(electric_field, iS)
```

Comment:
I think it would be better that this function takes the relevant config object. For many cases this might not be needed from the start. Also for certain scenarios one would like to get different propagation effects for the same ray tracing solution. This would not be possible with this interface.

Reply:
I think it is better to avoid passing the config as a function parameter. If it is stored as a class parameter, it is available for any class method. So if we want to make one of the functions configurable for one raytracer, we don't have to change the interface to do so.

Reply:
but this would mean calculating different attenuation options for the same ray become quite cumbersome. 
The interface change would be backward compatible. the function would just get an additional `config` attribute that could have a default setting. 
I agree, individual function arguments might limit us. but what about passing a config dict to the function?

Reply:
This wouldn't quite solve the problem. If we want to make another method of another raytracer configurable, we need to change the method call in simulation.py to pass the config as well. So unless we change the methods in all other raytracers to accept a config, we will get the "unrecognized parameter" error if they are used.

Reply:
no, the config dictionary that you would pass to the init function of the ray tracer, would just be passed to the function directly. If the dict contains an additional parameter that the function does not use, it won't complain.

Reply:
Yes, I am talking about passing the config to a function that did not accept it before. Let's say I want to make one of the other functions configurable in another raytracer. I would have to change that function in the raytracer to accept the config and change the simulation.py to pass the config to that function. If I don't change this function in all other raytracers, they will throw an error, since they now get an additional paramter they don't know.
As I said, I'm okay with implementing it this way, since I don't work on the raytracers much. But this does make the raytracers somewhat less modular

Reply:
I think we resolved this now with the get/set config method?